Configuration/Applications/python/cmsDriverOptions.py

```diff
@@ -229,10 +229,17 @@ def OptionsFromItems(items):
             options.isMC=True
         if 'SIM' in options.datatier:
             options.isMC=True
+        # This is a temporary solution to make validation-only sequence work in IB. One should define workflow properly.
+        if options.era and 'Phase2' in options.era:
+            options.isMC=True
+        if 'miniAODValidation' in options.step:
```

Comment:
It was counter-intuitive to me that all `miniAODValidation` must be considered as "MC": of course, we produce miniAOD also from real data.
Then I realized that the [`miniAODValidationSequence`](https://github.com/cms-sw/cmssw/blob/master/Validation/RecoParticleFlow/python/miniAODValidation_cff.py) only includes a `genParticlesValidation:` therefore, the fix here makes sense... but the name of the sequence probably not!
@hatakeyamak @jfernan2 : is there any (hystorical) reason for it?

Reply:
I've never seen that we use miniAODValidationSequence with data before. In UL, PdmV uses `DQM:@rerecoCommon` or `DQM:@standardDQM+@ExtraHLT`

I agree that this makes confusion on what it for MC/Data or both

[1, 2016 UL]
https://cmsweb.cern.ch/couchdb/reqmgr_config_cache/5ad31c9be7cfbf89c62a938d7fa94dfa/configFile

[2, B-Parking]
https://cmsweb.cern.ch/couchdb/reqmgr_config_cache/be67f91f945866144130b3b61e2e2994/configFile

Reply:
> > It was counter-intuitive to me that all `miniAODValidation` must be considered as "MC"
> 
> May I ask why? (I'm mostly just curious) I mean, we tend to use the word "validation" for MC and "DQM" for data.

Ah, ok: that was the key I missed

Reply:
> Can I simply say that if we have VALIDATION step, it will always be MC workflow?

This sounds pretty safe assumption.

> And if we have only DQM, it will be data workflow?

This is, on the other hand, less certain as DQM works (or should work) perfectly fine for MC as well. Counter-OTOH, the question is only for a default setting when neither data or MC is explicitly specified, so for the case where DQM is used without VALIDATION for MC one can explicitly set it.

To me it sounds a reasonable guess for default. The question is then if any IB test use cmsDriver with DQM without VALIDATION for MC.

Reply:
Then I assume we should be OK if DQM-only is used with MC because CMSSW should treat it like data in this way.

Reply:
> I mean, we tend to use the word "validation" for MC and "DQM" for data.

I believe this is not true in a practical and general way, as you can see from:

https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_steps.py

where there is a mixture of VALIDATION,DQM and VALIDATION/DQM over data and MC too

What it is true is that the DQM Unit Tests performed on:
https://github.com/cms-sw/cmssw/blob/master/DQMOffline/Configuration/test/runtests.sh#L23
are using a single (real) data input file, to simplify them, so we could switch these tests to isData=True to make them run with Phat's PR

In these tests, since a module with a given label can have different configurations depending on options such as Era,
Scenario, Data vs. MC etc, if multiple configurations for the same name were found, they are listed separately
here and denoted using subscripts. That is why the changes in this PR are failing somehow.

I don't know if the changes in this PR have implications in the standard workflows though

Reply:
> > I mean, we tend to use the word "validation" for MC and "DQM" for data.
> 
> I believe this is not true in a practical and general way, as you can see from:

Right, sorry for being imprecise. What I really meant (and what would have been sufficient) is that what we tend to call "validation" in the context of "standard sequences" requires MC.

Reply:
> What it is true is that the DQM Unit Tests performed on:
> https://github.com/cms-sw/cmssw/blob/master/DQMOffline/Configuration/test/runtests.sh#L23
> are using a single (real) data input file, to simplify them, so we could switch these tests to isData=True to make them run with Phat's PR
> 

This may not true. Somehow the test also picks DQM-only for Phase-2 workflow which does not exist (or I miss it somehow). CMSSW already guess that it is data-workflow, but still, fail. See in https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7e50b2/16755/unitTests/failed.html

Reply:
The input file for the tests is always the same, and it is an EGamma 2018 RAW data file downloaded at the beginning of the script to avoid delays from AAA:
--infile file:///data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-07-12-2300/tmp/004D960A-EA4C-E811-A908-FA163ED1F481.root